File: mydagger2/src/main/java/com/zhongjh/mydagger2/module/MainEntity.java
package com.zhongjh.mydagger2.module;

import javax.inject.Inject;

/**
 * @author zhongjh
 * @date 2022/4/24
 */
public class MainEntity {

    @Inject
    public MainEntity(String name) {
        this.name = name;
    }

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: myhilt/src/main/java/com/zhongjh/hiltdemo/havearguments/MainEntity.kt
package com.zhongjh.hiltdemo.havearguments

import dagger.assisted.Assisted
import dagger.assisted.AssistedInject

/**
 *
 * @author zhongjh
 * @date 2022/4/26
 */
class MainEntity @AssistedInject constructor(@Assisted("name") var name: String,@Assisted("area") var area: String)